cpgserver/src/main/scala/io/shiftleft/cpgservernew/route/CpgRoute.scala

```diff
@@ -0,0 +1,92 @@
+package io.shiftleft.cpgservernew.route
+
+import java.nio.file.{Files, Paths}
+import java.util.UUID
+
+import cats.effect.IO
+import cats.implicits.catsStdInstancesForList
+import cats.syntax.foldable._
+import io.circe.Encoder
+import io.circe.generic.auto._
+import io.circe.syntax._
+import org.http4s.circe._
+import org.http4s.dsl.io._
+import org.http4s.{EntityDecoder, HttpRoutes}
+
+final class CpgRoute(implicit cpgProvider: CpgProvider,
+                     cpgQueryExecutor: CpgQueryExecutor) {
+
+  import CpgRoute._
+
+  private def allFilesExist(fileNames: List[String]): IO[Boolean] = fileNames.forallM { fileName =>
```

Comment:
This method seems overlay long. Can we split it up into multiple smaller methods where names indicate higher-level operations?